internal/drivers/manager.go

```diff
@@ -514,7 +516,7 @@ func (m *Manager) setupInstance(ctx context.Context, pool *poolEntry, inuse bool
 	var inst *types.Instance
 
 	// generate certs
-	createOptions, err := certs.Generate(m.runnerName)
+	createOptions, err := m.certsCache.Get(m.runnerName)
```

Comment:
With this, it is possible that customer1 knows the certs for any build in 2 hr duration. Can the customer make rest calls to a different machine (of other account) using these certs?

Reply:
If we can ensure that east west traffic in the cloud vendor is not allowed, then we can have such restriction.

Reply:
nice catch @shubham149, i missed that. let me check